ledger-core/virtual/integration/vstatereport_test.go

```diff
@@ -92,3 +100,502 @@ func TestVirtual_SendVStateReport_IfPulseChanged(t *testing.T) {
 
 	require.Equal(t, 1, countVStateReport)
 }
+
+type stateReportCheckPendingCountersAndPulsesTestChecks struct {
+	UnorderedPendingCount         int32
+	UnorderedPendingEarliestPulse int // pulse 1-5
+	OrderedPendingCount           int32
+	OrderedPendingEarliestPulse   int // pulse 1-5
+}
+
+type stateReportCheckPendingCountersAndPulsesTestRequestInfo struct {
+	name  string
+	ref   reference.Global
+	flags payload.CallFlags
+	token payload.CallDelegationToken
+}
+
+type stateReportCheckPendingCountersAndPulsesTest struct {
+	mu sync.RWMutex
+
+	mc         *minimock.Controller
+	server     *utils.Server
+	runnerMock *logicless.ServiceMock
+
+	currentPulseIndex int
+	pulses            [6]pulse.Number
```

Comment:
why len is six and zero is unused?

Reply:
zero is used for "zero pulse number" (empty) and used in checks where needed

Reply:
added as many info as I could around value under index 0 in this array